tensorflow-ops/src/TensorFlow/Gradient.hs

```diff
@@ -687,9 +696,12 @@ opGrad "Fill" _ _ [dz] = [Nothing, Just $ sum dz rx]
   where
     rx = rangeOfRank dz
 
+opGrad "ReadVariableOp" _ _ [dz] = [Just $ expr dz]
```

Comment:
Please add a comment clarifying how this works, since it took me a bit to wrap my head around.

My understanding is: because the only input of `ReadVariableOp` is the variable itself, the gradient with respect to the read gets back propagated to the gradient w.r.t the variable.  And the variable node itself has no inputs, so no more backdrop takes place.